File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/C++/s378126458_commented.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int N, Q;
    cin >> N >> Q;
    string s;
    cin >> s;
    vector<int> pre(N+1, 0);

    for(int i = 1; i < N; i++) {
        if(s[i-1] == 'A' && s[i] == 'C') {
            pre[i] = 1;
        }
    }

    // Build prefix sum
    for(int i = 1; i < N; i++) {
        pre[i] += pre[i-1];
    }

    while(Q--) {
        int l, r;
        cin >> l >> r;
        r--;
        cout << (pre[r] - pre[l-1]) << endl;
    }
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s378126458_commented.go
package main

import (
	"fmt"
)

func main() {
	var N, Q int
	fmt.Scan(&N, &Q)
	var s string
	fmt.Scan(&s)
	pre := make([]int, N+1)
	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}
	// Build the prefix sum
	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}
	for q := 0; q < Q; q++ {
		var l, r int
		fmt.Scan(&l, &r)
		r--
		fmt.Println(pre[r] - pre[l-1])
	}
}

// End of Code
